screens/Discover.js

```diff
@@ -1,115 +1,83 @@
 import React, { Component } from "react";
 import DiscoverItem from "../components/DiscoverItem";
 import { MenuProvider } from "react-native-popup-menu";
-import { ScrollView, Text, StyleSheet } from "react-native";
+import { SafeAreaView, View, ScrollView, Text, StyleSheet } from "react-native";
+import getStyleSheet from "../styles/themestyles";
+import { ScreenStyles } from '../styles/global';
+
+import firebase from 'react-native-firebase';
 
 class Discover extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      darkTheme: window.darkTheme
+    }
+    this.workouts = []
+    firebase.database().ref('/common/workouts/').once('value').then((snapshot) => {
+      var wrks = []
+      snapshot.forEach(function (workout) {
+        wrks.push(workout.val())
+      });
+      this.workouts = wrks
+      this.setState({ dataReceived: true })
```

Comment:
You shouldn't call `setState` inside constructor. Move it to `componentDidMount()`

Reply:
As well as fetching data!

Reply:
Did you push changes? @AlexeiGitH

Reply:
Forgot. Just did.